File: src/features/typing/data/go/sort_colors.go
package main

func sortColors(nums []int) {
	low, mid, high := 0, 0, len(nums)-1
	for mid <= high {
		switch nums[mid] {
		case 0:
			nums[low], nums[mid] = nums[mid], nums[low]
			low++
			mid++
		case 1:
			mid++
		default:
			nums[mid], nums[high] = nums[high], nums[mid]
			high--
		}
	}
}

File: src/features/typing/data/dart/sort_colors.dart
class Solution {
  void sortColors(List<int> nums) {
    int low = 0;
    int mid = 0;
    int high = nums.length - 1;
    while (mid <= high) {
      if (nums[mid] == 0) {
        final tmp = nums[low];
        nums[low] = nums[mid];
        nums[mid] = tmp;
        low++;
        mid++;
      } else if (nums[mid] == 1) {
        mid++;
      } else {
        final tmp = nums[mid];
        nums[mid] = nums[high];
        nums[high] = tmp;
        high--;
      }
    }
  }
}
